### ISSUE TITLE
When wearing glasses with high correction, the closing button moves within the red circle

### ISSUE THREAD

**Issue by jbouter** [ubuntu/yaru#2798]
**Explanation/Summary**

When using the Yaru theme, the colour of the X within the close button differs from the background around the red circle. For people using glasses with high correction, this causes the X to move within the red circle based on the viewing angle. To resolve this issue, the colour of the X needs to be the same as the title bar background. Since this is the case when using yaru-light, that's a valid "workaround". If at all possible, without affecting your style and branding, I would like to be able to change the colour of the X within the red circle.

**Expected Behavior**

The closing button (X) remains in the centre of the closing button.

**Actual Behavior**

When wearing glasses, the closing button (X) moves within the red cricle.

**Steps to Reproduce the Problem**

This is going to seem silly:

1. Wear glasses
2. Use the default or dark yaru theme
3. look at the closing button and move your head from left to right

**Software that presents the issue**

All windows that use the yaru theme (and window borders)

**Yaru version**

* `20.04.10.1`
* `21.04.1`

Though, I'm fairly certain all version are affected.

**Screenshots**

Affected:

![Screenshot from 2021-04-23 18-38-04](https://user-images.githubusercontent.com/6340770/115902791-1a022d80-a463-11eb-8c8a-079db0bc33a5.png)

Not affected:

![Screenshot from 2021-04-23 18-38-11](https://user-images.githubusercontent.com/6340770/115902807-1d95b480-a463-11eb-9b6f-4eb7a1203bca.png)

**Upstream check**

This issue does not affect Adwaita, as Adwaita doesn't use a circle around the closing button and the colour of the X and background differ.

**Comment by jbouter** [comment]
If anyone could perhaps assist me with manually overriding the colour of the X in my own set-up, that would be great

**Comment by Muqtxdir** [comment]
Hi @jbouter 👋

As a person who wears glasses, I can acknowledge that this happens to me as well. Although to rectify this it would require some discussion on whether how to approach this with keeping the branding of Yaru. 

> If anyone could perhaps assist me with manually overriding the colour of the X in my own set-up, that would be great

In the meantime I could help you with overriding the window-close button color on your set-up. below is a mockup/screenshot similar to Adwaita based on your suggestion, please tell whether if this what you prefer for yourself so that I can put together a branch for you to try-out ?
![dark-after](https://user-images.githubusercontent.com/54065734/115990508-42179b00-a5e1-11eb-9d3d-270fa459c49f.png)

**Comment by jbouter** [comment]
Hi @Muqtxdir!

This indeed resolves the issue, and is close to Adwaita. But I do think it loses the recognisable Yaru branding.

Making the X dark also resolves the issue for me. I've made a mock-up in GIMP (possibly not using the right colours, but just to provide an example):

![dark_X](https://user-images.githubusercontent.com/6340770/115991921-2bfee000-a5cb-11eb-87b9-a8c9eab0776a.png)

**Comment by Feichtmeier** [comment]
I am really not a fan of the dark X
I would rather make the X only appear on hover then, like in macOS

**Comment by Feichtmeier** [comment]
@jbouter does this also happen in the gnome shell overview X-close button? (pres super or click on "activities" and then hover on a window preview)

**Comment by jbouter** [comment]
> I am really not a fan of the dark X
> I would rather make the X only appear on hover then, like in macOS

Yeah, I get that - or maybe have the background only appear (or change colour) on hover like the Breeze theme of the KDE desktop environment.

@jbouter does this also happen in the gnome shell overview X-close button? (pres super or click on "activities" and then hover on a window preview)

Yeah, it does. And in that case there's no difference between light or dark theme. But because the shell overview is only open when you're specifically looking for something, and you actively need to hover, it doesn't stand out as much.

**Comment by jbouter** [comment]
To add to the above comment. Using the gnome shell overview with the default Adwaita theme (where the appearing X has a blue background, rather than red) the effect of the X moving within the circle is less.

**Comment by Feichtmeier** [comment]
Some ideas

Cross only on hover
![grafik](https://user-images.githubusercontent.com/15329494/116006184-5ae87680-a60a-11eb-8581-9e4ce99a86a5.png)

![hover](https://user-images.githubusercontent.com/15329494/116006210-72276400-a60a-11eb-9ee2-2313528cc9dd.gif)

Cross only on hover with a black cross for close
![blackhover](https://user-images.githubusercontent.com/15329494/116006282-d4806480-a60a-11eb-895a-333ff1bb00ad.gif)

@ubuntu/yaru-team  wdyt?

**Comment by clobrano** [comment]
That's a clever idea, the only drawback is that we will be accused of copying MacOS 😁

**Comment by madsrh** [comment]
That is a pretty cool @Feichtmeier 
I would love to try that. Can you easily whip up a PR?

**Comment by Jupi007** [comment]
> Some ideas
> 
> Cross only on hover
> 
> Cross only on hover with a black cross for close
> 
> @ubuntu/yaru-team wdyt?

That looks very fresh @Feichtmeier
I prefer the white close icon version, the black one looks almost invisible.
As said @madsrh I'd love to try that in action 😃

**Comment by Feichtmeier** [comment]
Draft PR: https://github.com/ubuntu/yaru/pull/2807

:shrug: looks shockingly fresh, but then, is this okay? Can we just leave the symbols away? :D

**Comment by clobrano** [comment]
I'd keep them 😁, they're useful indication. Above all the minimize and maximize, since they'd be too similar otherwise

**Comment by elioqoshi** [comment]
> I would rather make the X only appear on hover then, like in macOS

The issue with that however is that that macOS has the same treatment for all 3 icons, not only the close button. It wouldn't make sense to keep minimize and expand the same while changing the close behaviour only, no?

> 
![hover](https://user-images.githubusercontent.com/15329494/116006210-72276400-a60a-11eb-9ee2-2313528cc9dd.gif)

This looks great! I'd just make the animations more faster in the implementation. However since this is a quite big change, we should test this out a bit further. I could do some user testing as well eventually.

**Comment by Feichtmeier** [comment]
User testing would be superb! @elioqoshi 

Those super slow animations come from a shitty gif. I have no idea why it is so slow on github xD I can assure you it isn't
Some valid points have also been raised on the PR which need to be fixed or we can't do it (firefox for example)

I'll keep you pinged when I think this is ready for user testing
